src/Components/Notifications/NotificationsList.tsx

```diff
@@ -65,8 +67,9 @@ const NotificationsList: React.FC<NotificationsListProps> = ({
       return
     }
 
-    setCurrentNotificationId(firstNotificationId)
-  })
+    router.replace(`/notification/${firstNotificationId}`)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
`NotificationContext` listens to URL changes and updates the notification ID automatically.